src/index.js

```diff
@@ -1,8 +1,74 @@
+const constants = require('./constants');
+const shellExec = require('shell-exec');
+
+function getProperties(type, addProps) {
+    return constants.PROPERTIES.any + 
+        (constants.PROPERTIES[type] ? ',' + constants.PROPERTIES[type] : '') + 
+        (addProps ? ',' + addProps : '');
+}
+
+function convertProperty(value) {
+    if (value === 'yes' || value === 'no')
+        return (value === 'yes');
+    
+    //if (/^[-+]?\d+$/.test(value))
+    //if (Number.isSafeInteger(value))
+    //    return parseInt(value);
+
+    //if (/[-+]?([0-9]*\.[0-9]+|[0-9]+)/.test(value))
+    const conv = parseFloat(value);
+    if (!Number.isNaN(conv))
+        return conv;
+    
+    return value;
+}
+
 /**
  * A module for the server-state system
- * @returns A JSON-serializable (via `JSON.stringify()`) version information about the server state
+ * 
+ * A module to view the current status of your systemd services
+ * 
+ * @throws if no units get passed to the function
+ * @throws if a unit gets passed two or more times
+ * 
+ * @argument options array of systemd units to check
+ * @returns {object|array|string|number|boolean} A JSON-serializable (via `JSON.stringify()`) version information about the server state
  */
-module.exports = async function() {
-    // TODO: Implement module
-    return 'Module loaded';
+module.exports = async function (units) {
+    if (!units || !units.length)
+        throw new Error('No units were specified!');
```

Comment:
Could we change this error message into something a little more "helpful"? E.g., `No units were specified. Expected argument to be of type string[], but got ${typeof units}`...

Also, we should avoid exclamation marks in error messages. They seem unnecessarily "loud"...

Reply:
Ok, good point. I will go through every Error message one more time.